libs/LedKit/source/LedKit.cpp

```diff
@@ -17,15 +29,44 @@ void LedKit::start(interface::LEDAnimation *animation)
 		return;
 	}
 
+	_event_flags.set(START_LED_ANIMATION_FLAG);
+}
+
+void LedKit::initializeAnimation()
+{
 	_animation->setLeds(_ears, _belt);
 	_animation->start();
+}
 
-	_animation_queue_id = _event_queue.call_every(40ms, _animation, &interface::LEDAnimation::run);
+void LedKit::runAnimation()
+{
+	_event_flags.clear(STOP_LED_ANIMATION_FLAG | END_OF_LED_ANIMATION_FLAG);
+
+	auto hasToStopLedAnimation = [&]() {
+		return (_event_flags.get() & STOP_LED_ANIMATION_FLAG) == STOP_LED_ANIMATION_FLAG;
+	};
+	while (!hasToStopLedAnimation() && !_animation->hasEnded()) {
+		_animation->run();
+		rtos::ThisThread::sleep_for(40ms);
+	}
+	if (_animation->hasEnded()) {
+		_event_flags.set(END_OF_LED_ANIMATION_FLAG);
+	}
+}
+
+void LedKit::run()
+{
+	while (true) {
+		_event_flags.wait_any(START_LED_ANIMATION_FLAG);
+
+		initializeAnimation();
```

Comment:
ici tu `initializeAnimation` à chaque itération du loop? ça me parait overkill et dangereux.